site/app/libraries/database/DatabaseQueries.php

```diff
@@ -5171,6 +5187,35 @@ public function getInstructorLevelAccessCourse(string $user_id): array {
         return $this->submitty_db->rows();
     }
 
+    /**
+     * @return array<Course>
+     */
+    public function getSelfRegistrationCourses(string $user_id): array {
+        $query = <<<SQL
+SELECT c.*, t.name AS term_name FROM courses c, terms t
+WHERE c.self_registration_type > ? AND c.status = 1 and c.course NOT IN (
```

Comment:
Is there a constant we can reference instead of the magic value 1 here?

Reply:
I don't believe that there currently a constant for the "Archived" status. Should I add one? (If so, in this PR, or in another PR?)

Reply:
i added the constant in this PR.